models/tshirt/include/Tshirt.h

```diff
@@ -0,0 +1,289 @@
+        //double column_total_soil_moisture_deficit;    //!< soil column total moisture deficit
+
+        tshirt_state(double ss, double sgw, double* nash_res_ptr = 0x0) : Ss(ss), Sgw(sgw), Snash(nash_res_ptr) {}
+    };
+
+    /*!
+     * Tshirt flux structure
+     *
+     * This structure provides storage for the fluxes generated by Tshirt at any time step
+     */
+    struct tshirt_fluxes
+    {
+        double surface_runoff;  //!< Direct surface runoff
+        double Qgw;             //!< Groundwater flow from groundwater reservoir to channel flow
+        double Qperc;           //!< Percolation flow from subsurface to groundwater reservoir
+        double Qlf;             //!< Lateral subsurface flow
+        double et_loss;         //!< Loss from ET
+
+        tshirt_fluxes(double q_gw, double q_perc, double q_lf, double runoff, double et_loss) : Qgw(q_gw),
+                                                                                                Qperc(q_perc),
+                                                                                                Qlf(q_lf),
+                                                                                                surface_runoff(runoff),
+                                                                                                et_loss(et_loss) {
+
+        }
+    };
+
+    // TODO: consider combining with or differentiating from similar hymod enum
+    enum TshirtErrorCodes
+    {
+        TSHIRT_NO_ERROR = 0,
+        TSHIRT_MASS_BALANCE_ERROR = 100
+    };
+
+    /*!
+     * Tshirt kernel class
+     *
+     * This class implements the Tshirt hydrological model.
+     */
+    class tshirt_kernel
+    {
+    public:
+
+        //! function to simulate losses due to evapotransportation
+        static double calc_et(double soil_m, void* et_params)
+        {
+            pdm03_struct* pdm = (pdm03_struct*) et_params;
+            pdm->XHuz = soil_m;
+            pdm03_wrapper(pdm);
+
+            return pdm->XHuz - soil_m;
+        }
+
+        //! Calculate the Cschaake, or the Schaake adjusted magic constant by soil type, based on the given Tshirt parameters
+        static double calc_Cschaake(const tshirt_params& params)
+        {
+            return 3.0 * params.satdk / (2.0e-6);
+        }
+
+        /*!
+         * Calculate the height above water table based on the given Tshirt parameters
+         *
+         * Calculate the height above water table based on known constants and the particular `alpha` value given for
+         * relative soil suction head, as provided in the Tshirt parameters.
+         *
+         * @param params
+         * @return
+         */
+        static double calc_Hwt(const tshirt_params& params)
+        {
+            // H_wt = alpha_fc * H_atm ; H_atm = P_atm / gamma ; P_atm = 101,300 [Pa] ; gamma = 9,810 [N m^-3] ('very nearly')
+            return params.alpha_fc * (101300 / 9810);
```

Comment:
Should define a constants header and macro these constants

Reply:
I've created `include/utilities/Constant.h` for this purpose.  It can be moved or renamed if necessary.